libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/oauth/parser/KeyStoreJwkKeyResolver.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.athenz.auth.oauth.parser;
+
+import java.net.URL;
+import java.security.Key;
+import java.util.concurrent.TimeUnit;
+import com.auth0.jwk.JwkException;
+import com.auth0.jwk.JwkProvider;
+import com.auth0.jwk.JwkProviderBuilder;
+import com.yahoo.athenz.auth.KeyStore;
+import com.yahoo.athenz.auth.util.AthenzUtils;
+import com.yahoo.athenz.auth.util.Crypto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwsHeader;
+import io.jsonwebtoken.SigningKeyResolver;
+
+/**
+ * KeyResolver that get public key from key store or JWKS URL
+ * @see <a href="https://tools.ietf.org/html/rfc7517" target="_top">RFC7517</a>
+ */
+public class KeyStoreJwkKeyResolver implements SigningKeyResolver {
```

Comment:
why did we create this class? we already have the SigningKeyResolver without relying on auth0 implementation and it supports both rsa and ec keys. So rather than creating yet another and adding dependency on auth0 library, take a look at the current key resolver and if necessary extend that class.

Reply:
As the auth0 class has built-in refresh logic on the JWKS URL, I included the library. However, its behavior is difficult to trace in operation so I updated to depend on the existing `JwtsSigningKeyResolver` in 591d10eb.

I also made some changes on the `JwtsSigningKeyResolver` class to support file URL.
In addition, as the 2 classes behave differently, I just use weak dependency. Please let me know if extending the class is better.